src/main/java/domain/Line.java

```diff
@@ -7,9 +7,16 @@
 public class Line {
 
     private final List<LineStatus> line;
+    private final LineGenerator lineGenerator;
 
-    public Line(int numberOfLine, LineGenerator lineGenerator) {
-        this.line = LineMaker.makeLine(lineGenerator, numberOfLine);
+    public Line(List<LineStatus> line, LineGenerator lineGenerator) {
+        this.line = line;
+        this.lineGenerator = lineGenerator;
+    }
+
+    public void addStatus(int numberOfLine) {
+        LineMaker.makeFirstLineStatus(lineGenerator, line);
+        LineMaker.makeElseLineStatus(lineGenerator, line, numberOfLine);
```

Comment:
addStatus를 외부에서 바라보았을 때, public void line.addStatus(int numberOfLine)을 볼 수 있는데요...!

이를 해석해보면 `line 클래스는 라인 수를 받아서 status를 추가한다`입니다.
그렇다면 내부적으로 status를 어떻게 추가하는지는 외부에 감추어야 할 하위 문제라고 할 수 있는데요.
객체의 인스턴스는 생성을 한 후에야 특정 기능의 역할을 할 수 있기때문에
전제조건으로 line은 List<LineStatus>, LineGenerator를 가지고 있다는 것이 고려되어야합니다. 

따라서, 생성자를 포함해서 해석해보면 
- Line은 List<LineStatus>, LineGenerator를 가지고 있고 이러한 속성을 사용해서 status를 추가한다.

라고 할 수 있습니다.

그런데 여기서 LineMaker를 의존해야할 필요가 있을까요??
오히려 아래와 같이
```suggestion
    public void addStatus(LineStatus lineStatus) {
        this.line.add(lineStatus);
    }
```
status를 받아서 처리하는 것 까지가 Line 클래스의 역할과 책임이고 line을 만드는 것은 다른 클래스의 역할과 책임이 아닐까요??🤔